cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/BufferingTable.java

```diff
@@ -951,6 +952,8 @@ public Row next() {
       } else {
         order = Bytes.compareTo(currentKey, currentRow.getRow());
       }
+
+      @SuppressWarnings("UnusedAssignment")
```

Comment:
Can't do that. javac is not smart enough to figure out that every branch of the if statement assign it. So you get an "uninitialized" error. But IntelliJ is smart enough and warns you about the unnecessary initializer :) It is personal preference: I prefer code that is free of warnings.